server/segment_rpc.go

```diff
@@ -386,9 +384,19 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 		return nil, err
 	}
 
-	// TODO(yondonfu):
-	// Set balUpdate.Debit = fee computed based on returned results
-	// Set balUpdate.Status = ReceivedChange
+	// We treat a response as "receiving change" where the change is the difference between the credit and debit for the update
+	balUpdate.Status = ReceivedChange
+	priceInfo := sess.OrchestratorInfo.PriceInfo
```

Comment:
Ah good catch. I think we can fix this by deferring execution of `updateOrchestratorInfo()` until the very end up of `SubmitSegment()` since the only current requirement is that the session's `OrchestratorInfo` is updated when `SubmitSegment()` returns

Reply:
Fixed in 9979bf4